server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -61,13 +66,24 @@ export class AssetController {
         const savedAsset = await this.assetService.createUserAsset(authUser, assetInfo, file.path, file.mimetype);
 
         if (uploadFiles.thumbnailData != null && savedAsset) {
```

Comment:
This makes sense, it should only check for the thumbnail is there or not, if there the process of saving an asset has a problem, it would throw an error in the `assetService`, and so this check won't be necessary.